packages/modules/order/src/services/order-module-service.ts

```diff
@@ -2302,6 +2302,137 @@ export default class OrderModuleService<
     return await this.revertLastChange_(order, sharedContext)
   }
 
+  @InjectManager()
+  async undoLastChange(
+    orderId: string,
+    @MedusaContext() sharedContext?: Context
+  ) {
+    const order = await super.retrieveOrder(
+      orderId,
+      {
+        select: ["id", "version"],
+      },
+      sharedContext
+    )
+
+    if (order.version < 2) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        `Order with id ${orderId} has no previous versions`
+      )
+    }
+
+    return await this.undoLastChange_(order, sharedContext)
+  }
+
+  @InjectTransactionManager()
+  protected async undoLastChange_(
+    order: OrderDTO,
+    sharedContext?: Context
+  ): Promise<void> {
+    const currentVersion = order.version
+
+    // Order Changes
+    const orderChanges = await this.orderChangeService_.list(
+      {
+        order_id: order.id,
+        version: currentVersion,
+      },
+      { select: ["id", "version"] },
+      sharedContext
+    )
+
+    const orderChangesIds = orderChanges.map((change) => {
+      return {
+        id: change.id,
+        status: OrderChangeStatus.PENDING,
```

Comment:
q: why are we setting it to `PENDING`?

In a scenario I tested, the order edit is `REQUESTED` and when I try to force confirm which fails due to low stock quantity the edit is reverted to pending instead of requested status. It's not a big deal from the admin standpoint since users can just continue the edit and rerequest again but just curious if we can revert the status as well?